src/compute/src/logging/compute.rs

```diff
@@ -375,16 +382,16 @@ pub fn construct<A: Allocate>(
         // Duration statistics derive from the non-rounded event times.
         let peek_duration = peek_duration
             .as_collection()
+            .explode_one(|(worker, val)| ((worker, val.next_power_of_two()), (val, 1)))
             // TODO(#16549): Use explicit arrangement
             .count_total_core()
-            .map({
-                move |((worker, pow), count)| {
-                    Row::pack_slice(&[
-                        Datum::UInt64(u64::cast_from(worker)),
-                        Datum::UInt64(pow.try_into().expect("pow too big")),
-                        Datum::UInt64(count),
-                    ])
-                }
+            .map(|((worker, bucket), (sum, count))| {
+                Row::pack_slice(&[
+                    Datum::UInt64(u64::cast_from(worker)),
+                    Datum::UInt64(bucket.try_into().expect("pow too big")),
+                    Datum::UInt64(count.try_into().expect("count too big")),
+                    u64::try_from(sum).ok().into(),
```

Comment:
Can we encode the specific `Datum` type here and above, like we do for the other columns?

Reply:
yes, but since it is nullable, it requires some verbosity:

```
u64::try_from(sum)
    .map(Datum::UInt64)
    .unwrap_or(Datum::Null)
```

Reply:
Well, the `.into()` is definitely less code (although it's also less clear what's happening, but up to you!)